src/main/java/com/vmware/vipclient/i18n/messages/api/opt/server/RemoteProductOpt.java

```diff
@@ -52,7 +52,7 @@ public ProductBasedOpt(BaseDTO dto) {
      * @see com.vmware.vipclient.i18n.messages.dao.ILocaleDao#getSupportedLocales()
      * @deprecated Replaced by {@link com.vmware.vipclient.i18n.messages.api.opt.server.RemoteLocaleOpt#getSupportedLanguages(String)()}
      */
-    @Deprecated public JSONArray getSupportedLocalesFromRemoteVIP() {
+    public List<String> getSupportedLocales() {
         JSONArray msgObject = new JSONArray();
         String responseStr = "";
         Map<String, Object> response = VIPCfg.getInstance().getVipService().getHttpRequester().request(V2URL.getSupportedLocaleListURL(
```

Comment:
Just like what I put in the Javadoc comment of this method,  the plan is to deprecate this V2URL endpoint on service side because it is but a subset of V2URL.getSupportedLanguageListURL (language tags only instead of tags + display names). So call RemoteLocaleOpt.getSupportedLanguages(default locale) in here and just grab the keySet().
```suggestion
    public List<String> getSupportedLocales() {
        Iterator<Locale> fallbackLocalesIter = LocaleUtility.getFallbackLocales().iterator();
        return getSupportedLocales(fallbackLocalesIter);
    }
    private List<String> getSupportedLocales(Iterator<Locale> fallbackLocalesIter) {
        if (fallbackLocalesIter != null && fallbackLocalesIter.hasNext()) {
            String locale = fallbackLocalesIter.next().toLanguageTag();
            Map<String, String> supportedLocales = RemoteLocaleOpt.getSupportedLanguages(locale);
            if (supportedLocales == null || supportedLocales.isEmpty()) {
                return getSupportedLocales(fallbackLocalesIter);
            }
            return supportedLocales.keySet();
        }
    }
    
```

Reply:
I prefer to deprecate this V2URL endpoint after V3 API is available.
Looks like V3 API url is different with V2, after V3 is available, this need to be changed again.